cmake/External/nccl.cmake

```diff
@@ -38,6 +42,7 @@ if(NOT __NCCL_INCLUDED)
       BUILD_IN_SOURCE 1
       CONFIGURE_COMMAND ""
       BUILD_COMMAND
+        env
```

Comment:
is this change intentional?

Reply:
Yep, it was there before, but removed by https://github.com/pytorch/pytorch/pull/83696. I thought we need to pass this to pass env vars to `make` if necessary?